### PR TITLE
Add vLLM on Ray microservice

### PR DESCRIPTION
**Pull request by XinyaoWa**
## Description

Add vLLM on Ray microservice

## Issues

#194 

## Type of change

- [ ] New feature (non-breaking change which adds new functionality)

### CODE REVIEW

**Comment by chensuyue** [comment]
Please add the e2e test like this one, https://github.com/opea-project/GenAIComps/blob/main/tests/test_llm_summarization_tgi.sh

File: comps/llms/text-generation/vllm_ray/vllm_openai.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        device = cli_args.pop("device")
+    else:
+        try:
+            from habana_frameworks.torch.distributed.hccl import initialize_distributed_hpu
+
+            initialize_distributed_hpu()
+            torch.zeros(1).to("hpu")
+            device = "HPU"
+        except Exception:
+            device = "GPU"
```
Comment (KepingYan) [inline comps/llms/text-generation/vllm_ray/vllm_openai.py:106]: This guide is for vllm on Guadi. We can remove this line and throw an Error to remind users to install the required environment.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Serve vLLM models with Ray Serve.", add_help=True)
+    parser.add_argument("--port_number", default=8008, type=int, help="Port number to serve on.")
+    parser.add_argument(
+        "--model_id_or_path", default="NousResearch/Meta-Llama-3-8B-Instruct", type=str, help="Model id or path."
+    )
+    parser.add_argument(
+        "--chat_processor", default="ChatModelNoFormat", type=str, help="Chat processor for aligning the prompts."
+    )
```
Comment (KepingYan) [inline comps/llms/text-generation/vllm_ray/vllm_openai.py:141]: It seems that this parameter is not used.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+
+# __serve_example_end__
+
+
+def main(argv=None):
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Serve vLLM models with Ray Serve.", add_help=True)
+    parser.add_argument("--port_number", default=8008, type=int, help="Port number to serve on.")
```
Comment (KepingYan) [inline comps/llms/text-generation/vllm_ray/vllm_openai.py:135]: Same here. Please confirm whether these two parameters can work normally as described in comps/llms/text-generation/vllm_ray/launch_vllm_ray.sh.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    #         },
+    #     ],
+    #     temperature=0.01,
+    #     stream=True,
+    # )
+
+    # for chat in chat_completion:
+    #     if chat.choices[0].delta.content is not None:
+    #         print(chat.choices[0].delta.content, end="")
+    # __query_example_end__
```
Comment (KepingYan) [inline comps/llms/text-generation/vllm_ray/vllm_openai.py:179]: It looks like we'll use microservice to send requests to server, please remove these if they are not needed here.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py (hunk 1 above)
Comment (XinyaoWa) [reply comps/llms/text-generation/vllm_ray/vllm_openai.py]: removed.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py (hunk 4 above)
Comment (XinyaoWa) [reply comps/llms/text-generation/vllm_ray/vllm_openai.py]: removed.

File: comps/llms/text-generation/vllm_ray/vllm_openai.py (hunk 2 above)
Comment (XinyaoWa) [reply comps/llms/text-generation/vllm_ray/vllm_openai.py]: removed.

**Pull request closed without merging**